### PR TITLE
Using permissions on buttons

### PR DESCRIPTION
**Pull request by tm1303**
## Context

Makes the edit panel button aware of the entity-links, which will drive the conditional rendering of the button itself. There are three results

1. PATCH is present, the button is rendered
2. PATCH is not present, the button is not rendered
3. The array of entity-links is null or undefined, in this case the button **is** rendered for legacy compatibility but this will be removed when all the entities have their links populated in the API

## Changes in this pull request

- EditPanelButton.tsx : now takes an option list of entity-links and uses this to decide whether to render itself
- EditPanelButton.test.tsx : added some tests

## Guidance to review

Kinda difficult to see in action as the API doesn't currently fulfil the second point (see above) but everywhere you could previously see an EditPanelButton you still can 😋

## Link to Trello card

https://trello.com/c/mpiONRQ6/478-edit-beacon-details-permissions

## Things to check

everywhere you could previously see an EditPanelButton you still can

### CODE REVIEW

File: src/components/dataPanel/EditPanelButton.tsx [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+}> = ({ onClick, links, children }): JSX.Element => {
+  const canEdit = links == null || links.some((link) => link.verb === "PATCH");
+  return canEdit ? (
+    <Typography align="right" style={{ paddingRight: "5em" }}>
+      <Button variant="outlined" onClick={onClick}>
+        {children}
+      </Button>
+    </Typography>
+  ) : (
+    <></>
```
Comment (zackads) [inline src/components/dataPanel/EditPanelButton.tsx:17]: I'm wondering if this is the best place to have the `allow user to edit` logic.  The `<></>` at line 17 makes me curious...

Why did you choose to put the logic here and not in the parent panel, for example?  I can think of pros/cons either way!  Keen to understand your thinking.

Comment (tm1303) [reply src/components/dataPanel/EditPanelButton.tsx:17]: it's in a single place here, if we put it above the button on the panel you'd have to repeat the logic for each panel types, BeaconPanel, BeaconUsePanel etc.

I think 🤔

Comment (zackads) [reply src/components/dataPanel/EditPanelButton.tsx:17]: Good point, works for me!

**Review by zackads — APPROVED** [review]
LGTM!

**Pull request merged**